pkg/util/podstore/podstore_test.go

```diff
@@ -0,0 +1,549 @@
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			pod, err := (&PodStore{}).checkDeletedPod(tt.obj)
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, tt.expectedResult, pod)
+			}
+		})
+	}
+}
+
+func Test_GetPodByIPAndTime(t *testing.T) {
+	k8sClient := fake.NewSimpleClientset()
+	podInformer := coreinformers.NewPodInformer(k8sClient, metav1.NamespaceAll, 0, cache.Indexers{})
+	tests := []struct {
+		name           string
+		ip             string
+		startTime      time.Time
+		expectedResult *v1.Pod
+	}{
+		{
+			name:           "no Pod in the Pod store",
+			ip:             "1.3.5.7",
+			startTime:      refTime,
+			expectedResult: nil,
+		},
+		{
+			name:           "find only one Pod in the Pod store - correct startTime",
+			ip:             "5.6.7.8",
+			startTime:      refTime.Add(-time.Minute),
+			expectedResult: pod2,
+		},
+		{
+			name:           "find only one Pod in the Pod store - incorrect startTime",
+			ip:             "5.6.7.8",
+			startTime:      refTime.Add(time.Minute),
+			expectedResult: pod2,
+		},
+		{
+			name:           "find current Pod in the Pod store",
+			ip:             "1.2.3.4",
+			startTime:      refTime.Add(time.Minute),
+			expectedResult: pod1,
+		},
+		{
+			name:           "find previous Pod in the Pod store",
+			ip:             "1.2.3.4",
+			startTime:      refTime.Add(-time.Minute),
+			expectedResult: pod4,
+		},
+		{
+			name:           "cannot find the Pod in the Pod store - SearchTime < CreationTime",
+			ip:             "1.2.3.4",
+			startTime:      refTime.Add(-time.Minute * 10),
+			expectedResult: nil,
+		},
+		{
+			name:           "cannot find the Pod in the Pod store - SearchTime > DeletionTime",
+			ip:             "1.2.3.4",
+			startTime:      refTime,
+			expectedResult: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			podStore := NewPodStore(podInformer)
+			err := podStore.pods.Add(pod1)
+			require.NoError(t, err)
+			err = podStore.pods.Add(pod2)
+			require.NoError(t, err)
+			err = podStore.pods.Add(pod4)
+			require.NoError(t, err)
+			podStore.timestampMap = timestampMap
+			pod, _ := podStore.GetPodByIPAndTime(tt.ip, tt.startTime)
+			assert.Equal(t, tt.expectedResult, pod)
+		})
+	}
+}
+
+func Test_processDeleteQueueItem(t *testing.T) {
+	fakeClock := clock.NewFakeClock(time.Now())
+	podStore := &PodStore{
+		pods:         cache.NewIndexer(podKeyFunc, cache.Indexers{podIPIndex: podIPIndexFunc}),
+		podsToDelete: workqueue.NewDelayingQueueWithCustomClock(fakeClock, deleteQueueName),
+		timestampMap: map[types.UID]*podTimestamps{"pod1": {}},
+	}
+	t.Run("delete Pod in podsToDelete/prevPod/timestampMap", func(t *testing.T) {
+		err := podStore.pods.Add(pod1)
+		require.NoError(t, err)
+		podStore.podsToDelete.Add(pod1)
+
+		result := podStore.processDeleteQueueItem()
+		assert.Equal(t, true, result)
+		if result {
```

Comment:
you should just use `require.Equal(...)`, and get rid of the conditional

Reply:
Fixed.